### PR TITLE
Revendor `github.com/gardener/machine-controller-manager@v0.48.1`

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
Revendor `github.com/gardener/machine-controller-manager@v0.48.1`. There is one breaking change (removed `.status.node` field which required manual adaptation).

**Special notes for your reviewer**:
/cc @himanshu-kun @rishabh-11 

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/gardener/dependency-watchdog v0.7.0
 	github.com/gardener/etcd-druid v0.15.3
 	github.com/gardener/hvpa-controller/api v0.5.0
-	github.com/gardener/machine-controller-manager v0.45.0
+	github.com/gardener/machine-controller-manager v0.48.1
```
Comment (ialidzhikov) [inline go.mod:14]: Aren't there any changes to the MCM CRDs (currently maintained under `extensions/pkg/controller/worker/tempaltes/crds`)?

I think we should also put comments on these MCM CRDs from which source they are maintained. For example https://github.com/gardener/gardener/blob/a243225097a891aa497f70bdf125424bd5cd55d3/pkg/operation/botanist/component/etcd/crds/templates/crd-druid.gardener.cloud_etcdcopybackupstasks.yaml#L1, https://github.com/gardener/gardener/blob/9b4bee5e9999a486f99e313c903912c4bc8d99d7/pkg/operation/botanist/component/vpa/templates/crd-verticalpodautoscalercheckpoints.tpl.yaml#L2. 
Then you know which version of the CRDs you use.

Comment (rfranzke) [reply go.mod:14]: I added f56ca6f7add68de39a42dc89c2016be61aa5c689 and 468e5624d7b106e037188f0194306485fdf7d062, PTAL

**Comment by ialidzhikov** [comment]
My first assumption for the failing tests is that machine-controller-manager-provider local has to vendor machine-controller-manager >= v0.48.0 (currently vendors v0.47.0) to adopt the changes from https://github.com/gardener/machine-controller-manager/commit/4fba966be64ce37bbf242be27aae49ae2f3061ba.

**Comment by ialidzhikov** [comment]
I tried to look into https://github.com/gardener/gardener/pull/7599#issuecomment-1457897867. I hit other issues in my local setup that I have to debug further:
```
{"level":"error","ts":"2023-03-07T16:44:20.071+0200","logger":"shoot-webhook.handlerShootClient","msg":"Admission denied","kind":"Node","namespace":"","name":"machine-shoot--local--local-local-7bdd8-w7n87","error":"could not process: could not find shoot namespace for webhook request"
```

For the particular issue, I think the best would be to revert the CRD update (https://github.com/gardener/gardener/pull/7599/commits/468e5624d7b106e037188f0194306485fdf7d062). With the CRD update the `status.node` is removed and I guess this will break all provider extensions with MCM versions < 0.48 and MCM providers vendoring MCM < 0.48. @himanshu-kun @rishabh-11 feel free to correct me if I am wrong. 

When I asked for CRD update, I didn't know that there is such breaking change.

**Comment by himanshu-kun** [comment]
> I guess this will break all provider extensions with MCM versions < 0.48 and MCM providers vendoring MCM < 0.48. @himanshu-kun @rishabh-11 feel free to correct me if I am wrong.

 I agree with @ialidzhikov we should keep the old CRDs(with status.Node) intact for now, so that the machineControllers, extension providers in landscapes depending on `status.Node` don't break.
I had thought of updating these CRDs and remove `status.Node` myself later, after I was sure all controllers in landscapes are updated and not dependant on `status.Node`.
refer https://github.com/gardener/machine-controller-manager-provider-alicloud/pull/40#issuecomment-1441250100

File: extensions/pkg/controller/worker/predicate.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func MachineStatusHasChanged() predicate.Predicate {
 		}
 
 		return oldMachine.Spec.ProviderID != newMachine.Spec.ProviderID ||
-			oldMachine.Status.Node != newMachine.Status.Node
+			oldMachine.Labels[machinev1alpha1.NodeLabelKey] != newMachine.Labels[machinev1alpha1.NodeLabelKey]
```
Comment (himanshu-kun) [inline extensions/pkg/controller/worker/predicate.go:39]: should the function Name change ?

File: extensions/pkg/controller/worker/predicate_test.go [hunk 3]
```diff
@@ -55,7 +56,7 @@ var _ = Describe("Worker Predicates", func() {
 
 		It("should notice the change of the Node in the Status", func() {
 			predicate := worker.MachineStatusHasChanged()
-			newMachine.Status.Node = "ip.10-256-18-291.cluster.node"
+			metav1.SetMetaDataLabel(&newMachine.ObjectMeta, "node", "ip.10-256-18-291.cluster.node")
```
Comment (himanshu-kun) [inline extensions/pkg/controller/worker/predicate_test.go:68]: test description should change, in my opinion

File: extensions/pkg/controller/worker/predicate_test.go [hunk 4]
```diff
@@ -64,8 +65,8 @@ var _ = Describe("Worker Predicates", func() {
 
 		It("should not react when there are no changes of the Node in the Status", func() {
 			predicate := worker.MachineStatusHasChanged()
-			oldMachine.Status.Node = "ip.10-256-18-291.cluster.node"
-			newMachine.Status.Node = "ip.10-256-18-291.cluster.node"
+			metav1.SetMetaDataLabel(&oldMachine.ObjectMeta, "node", "ip.10-256-18-291.cluster.node")
```
Comment (himanshu-kun) [inline extensions/pkg/controller/worker/predicate_test.go:77]: same as above

**Review by himanshu-kun — CHANGES_REQUESTED** [review]
requested name changes, let me know if they make sense.

**Review by ialidzhikov — COMMENTED** [review]
/approve

**Review by himanshu-kun — APPROVED** [review]
/lgtm

**Pull request merged**